dropwizard-service-discovery-bundle/src/main/java/io/appform/dropwizard/discovery/bundle/id/IdGenerator.java

```diff
@@ -209,45 +194,35 @@ public static Id generate(String prefix) {
         }
     }
 
-    @Data
-    private static class GenerationResult {
-        private final Id id;
-        private final IdValidationState state;
-    }
     /**
      * Generate id that mathces all passed constraints.
      * NOTE: There are performance implications for this.
      * The evaluation of constraints will take it's toll on id generation rates. Tun rests to check speed.
      *
      * @param prefix        String prefix
      * @param inConstraints Constraints that need to be validate.
-     * @param skipGlobal Skip global constrains and use only passed ones
+     * @param skipGlobal    Skip global constrains and use only passed ones
      * @return Id if it could be generated
      */
     public static Optional<Id> generateWithConstraints(String prefix, final List<IdValidationConstraint> inConstraints, boolean skipGlobal) {
-        try {
-            final GenerationResult generationResult = retrier.call(() -> {
-                Id id = generate(prefix);
-                return new GenerationResult(id, validateId(inConstraints, id, skipGlobal));
-            });
-            return Optional.ofNullable(generationResult.getId());
-        }
-        catch (ExecutionException e) {
-            log.error("Error occurred while generating id with prefix " + prefix, e);
-        }
-        catch (RetryException e) {
-            log.error("Failed to generate id with prefix " + prefix + " after max attempts (512)", e);
+        val generationResult = RETRIER.get(
```

Comment:
Return Optional.filter().map() of this instead of the if-else chain

Reply:
Done